src/gui/torrentcontentmodel.cpp

```diff
@@ -41,27 +56,113 @@
 
 namespace
 {
+    const char showSystemFileIconsSettingKey[] = "Preferences/Advanced/showSystemFileIcons";
+
     QIcon getDirectoryIcon()
     {
         static QIcon cached = GuiIconProvider::instance()->getIcon("inode-directory");
         return cached;
     }
 
-    QIcon getFileIcon()
+    class UnifiedFileIconProvider: public QFileIconProvider
     {
-        static QIcon cached = GuiIconProvider::instance()->getIcon("text-plain");
-        return cached;
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            Q_UNUSED(info);
+            static QIcon cached = GuiIconProvider::instance()->getIcon("text-plain");
+            return cached;
+        }
+    };
+#ifdef Q_OS_WIN
+    // See QTBUG-25319 for explanation why this is required
+    class WinShellFileIconProvider: public UnifiedFileIconProvider
+    {
+    public:
+        using QFileIconProvider::icon;
+        QIcon icon(const QFileInfo &info) const override
+        {
+            SHFILEINFO sfi = { 0 };
+            HRESULT hr = ::SHGetFileInfoW(info.absoluteFilePath().toStdWString().c_str(),
```

Comment:
>Maybe we should do this initialization in qBt globally, in the Application class?

I doubt it since:
>CoInitialize function
>Initializes the COM library on the current thread ...

It's thread local.